pulp_smash/tests/rpm/api_v2/test_sync_publish.py

```diff
@@ -0,0 +1,650 @@
+    :param href: A string. The path to the repository on which the association
+        action is being performed. Content is copied to this repo.
+    :param responses: A list, or some other object that supports the ``append``
+        method. If given, all server responses are appended to this object.
+    :returns: The server's JSON-decoded response.
+
+    """
+    return _handle_response(requests.post(
+        server_config.base_url + href + 'actions/associate/',
+        json={'source_repo_id': source_repo_id},
+        **server_config.get_requests_kwargs()
+    ), responses)
+
+
+def _add_yum_distributor(server_config, href, responses=None):
+    """Add a yum distributor to an RPM repository.
+
+    The yum distributor will not automatically be published, and is available
+    over HTTP and HTTPS. The distributor ID and relative URL are random.
+
+    :param server_config: A :class:`pulp_smash.config.ServerConfig` object.
+    :param href: A string. The path to the repository to which a distributor
+        shall be added.
+    :param responses: A list, or some other object that supports the ``append``
+        method. If given, all server responses are appended to this object.
+    :returns: The server's JSON-decoded response.
+
+    """
+    return _handle_response(requests.post(
+        server_config.base_url + href + 'distributors/',
+        json={
+            'auto_publish': False,
+            'distributor_id': rand_str(),
+            'distributor_type_id': 'yum_distributor',
+            'distributor_config': {
+                'http': True,
+                'https': True,
+                'relative_url': '/' + rand_str(),
+            },
+        },
+        **server_config.get_requests_kwargs()
+    ), responses)
+
+
+def _publish_repo(server_config, href, distributor_id, responses=None):
+    """Publish a repository.
+
+    :param server_config: A :class:`pulp_smash.config.ServerConfig` object.
+    :param href: A string. The path to the repository to which a distributor
+        shall be added.
+    :param distributor_id: The ID of the distributor performing the publish.
+    :param responses: A list, or some other object that supports the ``append``
+        method. If given, all server responses are appended to this object.
+    :returns: The server's JSON-decoded response.
+
+    """
+    return _handle_response(requests.post(
+        server_config.base_url + href + 'actions/publish/',
+        json={'id': distributor_id},
+        **server_config.get_requests_kwargs()
+    ), responses)
+
+
+def _gen_rpm_repo_body():
+    """Return a semi-random dict that can be used for creating an RPM repo."""
+    return {
+        'id': rand_str(),
+        'importer_config': {},
+        'importer_type_id': 'yum_importer',
+        'notes': {'_repo-type': 'rpm-repo'},
+    }
+
+
+class _TaskTimedOutException(Exception):
+    """Indicates that polling a task timed out."""
+
+
+def _poll_tasks(server_config, href):
```

Comment:
Would it be better to put this in some place that is shared by other tests? We need to poll tasks for other plugins also.